lib/rubocop/cop/lint/interpolated_string.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for interpolation in a single quoted string.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #  foo = 'something with #{interpolation} inside'
+      #
+      # @example
+      #
+      #   # good
+      #
+      #  foo = "something with #{interpolation} inside"
+      class InterpolationCheck < Cop
+        MSG = 'Did you want to interpolate?'.freeze
```

Comment:
Hm. Maybe:

> Interpolation in single quoted string detected. Use double quoted strings if you need interpolation.

The first part could be removed when we add the improved highlighting. 🙂 WDYT?